### PR TITLE
[chrome] Promisify remaining `chrome.identity` APIs

### PR DESCRIPTION
**Pull request by fregante**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.chrome.com/docs/extensions/reference/identity/#method-getAccounts
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/chrome/index.d.ts [hunk 1]
```diff
@@ -4790,30 +4790,35 @@ declare namespace chrome.identity {
      *  * Removes all OAuth2 access tokens from the token cache
      *  * Removes user's account preferences
      *  * De-authorizes the user from all auth flows
+     * If `callback` is not provided, the function returns a Promise when the state has been cleared.
      * @since Chrome 87.
      * @param callback Called when the state has been cleared.
      * The parameter should be a function that looks like this:
      * () => {...};
```
Comment (fregante) [inline types/chrome/index.d.ts:4797]: These lines are redundant and I'll probably send a PR in the future to remove it. TSDoc should not repeat types

File: types/chrome/index.d.ts [hunk 2]
```diff
@@ -4831,12 +4836,14 @@ declare namespace chrome.identity {
...
      * If an access token is discovered to be invalid, it should be passed to removeCachedAuthToken to remove it from the cache. The app may then retrieve a fresh token with getAuthToken.
+     * If `callback` is not provided, the function returns a Promise when the state has been removed from the cache.
      * @param details Token information.
      * @param callback Called when the token has been removed from the cache.
      * If you specify the callback parameter, it should be a function that looks like this:
      * function() {...};
      */
-    export function removeCachedAuthToken(details: TokenInformation, callback?: () => void): void;
+    export function removeCachedAuthToken(details: TokenInformation): Promise<void>;
+    export function removeCachedAuthToken(details: TokenInformation, callback: () => void): void;
```
Comment (fregante) [inline types/chrome/index.d.ts:4846]: Also in the future: there should not be a `callback?` parameter, but just `callback` and a signature that returns a `Promise`. 

Many "promisified" APIs in this file incorrectly assert that if `callback` is not provided, the function still returns `void`

**Review by fregante — COMMENTED** [review]
This PR follows:

- https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64135

_Personal reference:_

- _written for and tested in https://github.com/pixiebrix/pixiebrix-extension/pull/5150_

**Review by matcarlson — APPROVED** [review]
Thanks for your updates.

**Comment by fregante** [comment]
Ready to merge

**Pull request merged**